pkg/clone/clone.go

```diff
@@ -0,0 +1,152 @@
+package clone
+
+import (
+	"encoding/json"
+	"fmt"
+
+	//nolint:staticcheck
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"path"
+	"path/filepath"
+)
+
+type ScriptStruct struct {
+	path      string
+	arguments []string
+}
+
+type Clone struct {
+	URL string
+}
+
+type Response struct {
+	CID string
+}
+
+func NewCloneClient() (*Clone, error) {
+	return &Clone{
+		URL: "",
+	}, nil
+}
+
+func (cl *Clone) RunShellScript(scriptPath string, args []string) (string, error) {
+	scriptArgs := []string{}
+	scriptArgs = append(scriptArgs, scriptPath)
+	scriptArgs = append(scriptArgs, args...)
+
+	if len(args) > 0 {
+		last := scriptArgs[len(scriptArgs)-1]
+		path, err := ioutil.TempDir("", path.Base(last))
+		scriptArgs[len(scriptArgs)-1] = path
+		if err != nil {
+			fmt.Println(err)
+		}
+		return path, nil
+	}
+
+	x := fmt.Sprintf("/bin/bash %v", scriptArgs)
+	fmt.Print(x)
+	cmd := exec.Command(x)
+	err := cmd.Start()
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Printf("Waiting for %s script to finish...", scriptPath)
+	err = cmd.Wait()
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println("Script finished")
+	// Run the command and get the output
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Println()
+		return "", fmt.Errorf("failed to execute script: %s", err)
+	}
+	fmt.Println()
+	// Print the output
+	fmt.Printf("%s\n", output)
+	return "", nil
+}
+
+func (cl *Clone) IfNotInstalledInstallingGitlfs() {
+	_, err := filepath.Abs("./install-lfs.sh")
```

Comment:
where are you reading these scripts from? Are we trying to install dependencies at runtime on behalf of the user?

Reply:
That's unused code my linter didn't catch it I think
I had that previously but since sudo didn't work with go 

Will remove that